src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -189,6 +197,41 @@ protected Probe createProbe(Integer externalPort, String endpoint, int timeout,
 		return cmdArgs;
 	}
 
+	/**
+	 * Volume mount deployment properties are specified in the following comma separated format:
+	 *
+	 * <code>
+	 *     spring.cloud.deployer.kubernetes.volumeMounts=name:path[:true],name:path[:true]
+	 * </code>
+	 *
+	 * Where <code>name</code> is the name of the volume mount and should match a configured Volume,
+	 * <code>path</code> is the mount path (E.g. <code>/tmp/inside/container</code>) and optionally the
+	 * readonly flag (default is <code>false</code> if not specified).
+	 *
+	 * Volume mounts can be specified as deployer properties as well as app deployment properties.
+	 * Deployment properties override deployer properties.
+	 *
+	 * @param request
+	 * @return the configured volume mounts
+	 */
+	protected List<VolumeMount> getVolumeMounts(AppDeploymentRequest request) {
+		Set<VolumeMount> volumeMounts = new HashSet<>();
+
+		String volumeMountDeploymentProperty = request.getDeploymentProperties()
+				.getOrDefault("spring.cloud.deployer.kubernetes.volumeMounts", "");
+		if (!StringUtils.isEmpty(volumeMountDeploymentProperty)) {
+			String[] volumePairs = volumeMountDeploymentProperty.split(",");
+			for (String volumePair : volumePairs) {
+				String[] volume = volumePair.split(":");
+				Assert.isTrue(volume.length <= 3, format("Invalid volume mount: '{}'", volumePair));
+				volumeMounts.add(new VolumeMount(volume[1], volume[0],
+						volume.length == 3 ? Boolean.valueOf(volume[2]) : Boolean.FALSE, null));
+			}
+		}
+		properties.getVolumeMounts();
```

Comment:
I don't think this statement does anything, should't this be the returned if `volumeMountDeploymentProperty` is empty?